docs/decisions/0004-downstream-effect-events.rst

```diff
@@ -0,0 +1,103 @@
+without needing a response as one would in a REST setup.
+
+This is being done to circumvent messy circular dependencies, wherein edx-exams and edx-platform go back and
+forth asking each other for information.
+
+
+Decision
+--------
+We are porting these because they are both essential and easy to translate into events:
+
+  * Grades Override - A call to the grades service to override a grade when an exam attempt is rejected.
+
+  * Undo Grades Override - A call to the grades to undo such an override when marking a rejected attempt as verified.
+
+  * Instructor Delete Attempt - A call to the instructor service to delete an exam attempt.
+
+  * Instructor Complete Attempt - A call to the instructor service that marks the exam as completed.
+
+  * Instructor Is Course Staff - A call to the instructor service to check if a user is course staff.
+
+  * Invalidate Certificate - A call to the certificates service to invalidate a leaner’s edx certificate for a course.
+
+  * Set Credit Requirement Status - A call to the credits service to create or modify a learner’s credit requirement status for an exam.
+
+Here’s what we’re NOT porting (and why):
+
+  * Credit Prerequisite Check:
+
+    * In edx-proctoring, we call the credits service to see if a learner has completed the prerequisites for an exam.
+
+    * Translating this into edx-exams would require a request-response call to be made from edx-exams to edx-platform. This would create an undesired circular dependency, as we already have edx-platform making such calls to edx-exams.
+
+    * Instead, we have decided to implement an endpoint in the credits service that returns the prereq status, which will be called directly from the exams UI.
+
+    * This feature will be developed separately and is considered to be outside of the scope of M6.
+
+  * Name Affirmation:
+
+    * Currently, edx-proctoring calls the name affirmation service in order to match the name that Proctortrack sees on the learner’s ID to the user’s verified name in the LMS database.
+
+    * We do not plan to implement this since It is very likely that 2U will not require the matching of one’s verified name in edx-platform to the names on their IDs in the future.
+
+    * Furthermore, it will likely take a very significant amount of time for Verificient to implement ID Verification in LTI form, so we do not need to do this until we know that they are implementing that feature on their end.
+
+    * Ideally, we want to be able to call the name affirmation service to inject the user’s first and last name into the LTI launch data data as the “User’s Verified Name”, but the logic regarding verified names isn’t even coded out in xblock-lti-consumer because the value of such names are unclear right now.
+
+  * Enrollments:
+
+    * This is only called to get a learner’s onboarding status.
+
+    * We will NOT be implementing this, since we are not implementing onboarding exams to reach MVP.
```

Comment:
You will not be implementing these as part of this project. But someone else can come a long and implement. Therefore the language should be:
"The following are not implemented as part of this decision".